### ISSUE TITLE
Intrepid2: MonolithicExecutable unit test fails in Cuda builds on Power9+Volta70 arch

### ISSUE THREAD

**Issue by ndellingwood** [trilinos/Trilinos#12037]
## Bug Report
@trilinos/intrepid2 

The `Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests_MPI_1` test fails in Cuda builds (with and without UVM enabled) in the `StructuredIntegration_HcurlFormulation_UniformAlgorithm_D2_P1_QuadratureUniformMesh_UnitTest` subtest :

```bash
220. StructuredIntegration_HcurlFormulation_UniformAlgorithm_D2_P1_QuadratureUniformMesh_UnitTest ... [weaver4:3206220:0:3206220] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
==== backtrace (tid:3206220) ====
 0  /home/projects/ppc64le-pwr9-nvidia/spack-installs/ucx/1.12.1/gcc/8.3.1/base/z5nsbpl/lib/libucs.so.0(ucs_handle_error+0x374) [0x20002dd56ea4]
 1  /home/projects/ppc64le-pwr9-nvidia/spack-installs/ucx/1.12.1/gcc/8.3.1/base/z5nsbpl/lib/libucs.so.0(+0x37070) [0x20002dd57070]
 2  /home/projects/ppc64le-pwr9-nvidia/spack-installs/ucx/1.12.1/gcc/8.3.1/base/z5nsbpl/lib/libucs.so.0(+0x374a0) [0x20002dd574a0]
 3  linux-vdso64.so.1(__kernel_sigtramp_rt64+0) [0x2000000604d8]
 4  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x11e21b74]
 5  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x116e887c]
 6  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x11708ff0]
 7  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x1170b064]
 8  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x118c6670]
 9  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x118c71ac]
10  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x12018284]
11  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x1201da54]
12  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x1201f06c]
13  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x12020340]
14  /home/ndellin/trilinos/Trilinos-pristine/Build/Weaver-Cuda11-Gcc830-nightly-nouvm-select-packages/packages/intrepid2/unit-test/MonolithicExecutable/Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests.exe() [0x10100f70]
15  /lib64/glibc-hwcaps/power9/libc-2.28.so(+0x29de8) [0x20002d439de8]
16  /lib64/glibc-hwcaps/power9/libc-2.28.so(__libc_start_main+0xb4) [0x20002d439fd4]
```

### Steps to Reproduce
1. SHA1: 34934050ffee2ffe5ad75d1f08c12697cfd6c1f6
2. Configure script: (For Weaver testbed, rhel8 queue):

```bash
# Get interactive a compute node
bsub -Is -n 1 -q rhel8 -gpu "num=1" bash

# Load env
export ATDM_CONFIG_REGISTER_CUSTOM_CONFIG_DIR=${TRILINOS_DIR}/cmake/std/atdm/contributed/weaver
source ${TRILINOS_DIR}/cmake/std/atdm/load-env.sh weaver-cuda-11.2-opt
export OMPI_CXX="$KOKKOS_PATH/bin/nvcc_wrapper"

# Configuration - Cuda build, no UVM
cmake \
      -DCMAKE_CXX_FLAGS='-g' \
      -DCMAKE_CXX_STANDARD="17" \
      -DCMAKE_INSTALL_PREFIX=$PWD/install \
      -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
      -DTrilinos_ENABLE_COMPLEX_DOUBLE=ON \
      -DTrilinos_ENABLE_TESTS=OFF \
      -DTrilinos_ENABLE_ALL_PACKAGES=OFF \
      -DTPL_ENABLE_CUSPARSE:BOOL=ON \
      -DFC_FN_UNDERSCORE=UNDER \
      \
      -D Trilinos_ENABLE_Kokkos=ON \
      -D Kokkos_ARCH_VOLTA70=ON \
      -D Kokkos_ARCH_POWER9=ON \
      -D Kokkos_ENABLE_CUDA=ON \
      -D Kokkos_ENABLE_CUDA_LAMBDA=ON \
      -D Kokkos_ENABLE_CUDA_UVM=OFF \
      -DTrilinos_ENABLE_Sacado=ON \
      -DTrilinos_ENABLE_Intrepid2=ON \
      -DIntrepid2_ENABLE_TESTS=ON \
      \
$TRILINOS_DIR

```

**Comment by CamelliaDPG** [comment]
Thanks, @ndellingwood, for the report and for including reproduction instructions!  I'll take a look.

**Comment by ndellingwood** [comment]
Thanks @CamelliaDPG !

**Comment by ndellingwood** [comment]
Resolved by #12047
